lib/llhttp/parser.js

```diff
@@ -0,0 +1,261 @@
+'use strict'
+
+/* global WebAssembly */
+
+const { resolve } = require('path')
+const { readFileSync } = require('fs')
+const constants = require('./constants')
+const WASM_BUILD = resolve(__dirname, './llhttp.wasm')
+const bin = readFileSync(WASM_BUILD)
+const mod = new WebAssembly.Module(bin)
+
+const REQUEST = constants.TYPE.REQUEST
+const RESPONSE = constants.TYPE.RESPONSE
+const kOnMessageBegin = 0
+const kOnHeaders = 1
+const kOnHeadersComplete = 2
+const kOnBody = 3
+const kOnMessageComplete = 4
+
+const kPtr = Symbol('kPrt')
+const kUrl = Symbol('kUrl')
+const kStatusMessage = Symbol('kStatusMessage')
+const kHeadersFields = Symbol('kHeadersFields')
+const kHeadersValues = Symbol('kHeadersValues')
+const kBody = Symbol('kBody')
+const kGetHeaders = Symbol('kGetHeaders')
+const kReset = Symbol('kReset')
+const kMakeError = Symbol('kMakeError')
+
+const cstr = (ptr, len) =>
+  Buffer.from(inst.exports.memory.buffer, ptr, len).toString()
+
+/* eslint-disable camelcase */
+const wasm_on_message_begin = p => {
+  const i = instMap.get(p)
+  i[kReset]()
+  return i[kOnMessageBegin]()
+}
+
+const wasm_on_url = (p, at, length) => {
+  instMap.get(p)[kUrl] = cstr(at, length)
+  return 0
+}
+
+const wasm_on_status = (p, at, length) => {
+  instMap.get(p)[kStatusMessage] = cstr(at, length)
+  return 0
+}
+
+const wasm_on_header_field = (p, at, length) => {
+  const i = instMap.get(p)
+  i[kHeadersFields].push(cstr(at, length))
+  return 0
+}
+
+const wasm_on_header_value = (p, at, length) => {
+  const i = instMap.get(p)
+  i[kHeadersValues].push(cstr(at, length))
+  return 0
+}
+
+const wasm_on_headers_complete = p => {
+  const i = instMap.get(p)
+  const type = inst.exports.llhttp_get_type(p)
+  const versionMajor = inst.exports.llhttp_get_http_major(p)
+  const versionMinor = inst.exports.llhttp_get_http_minor(p)
```

Comment:
We don't need the version stuff I think?

Reply:
Not sure, it seems like it though.